File: backend/app/schemas/audit.py
"""Schémas explicites de consultation du journal d'audit."""

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class AuditResourceLink(BaseModel):
    label: str
    resource_type: str
    resource_id: uuid.UUID
    route: str


class AuditEventOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    cabinet_id: uuid.UUID
    entreprise_id: uuid.UUID | None = None
    entreprise_nom: str | None = None
    user_id: uuid.UUID | None = None
    actor_name: str | None = None
    actor_email: str | None = None
    actor_role: str | None = None
    actor_type: str
    action: str
    module: str
    resource_type: str | None = None
    resource_id: uuid.UUID | None = None
    description: str | None = None
    status: str
    event_at: datetime
    correlation_id: str | None = None
    item_count: int | None = None
    resource_ids: list[str] = Field(default_factory=list)
    old_values: dict[str, Any] | None = None
    new_values: dict[str, Any] | None = None
    metadata: dict[str, Any] | None = None
    links: list[AuditResourceLink] = Field(default_factory=list)


class AuditPageOut(BaseModel):
    items: list[AuditEventOut]
    page: int
    page_size: int
    total: int
    pages: int


class AuditFilterOptionsOut(BaseModel):
    actions: list[str]
    modules: list[str]
    roles: list[str]
    statuses: list[str]

File: frontend/src/types/audit.ts
export interface AuditResourceLink {
  label: string;
  resource_type: string;
  resource_id: string;
  route: string;
}

export interface AuditEvent {
  id: string;
  cabinet_id: string;
  entreprise_id: string | null;
  entreprise_nom: string | null;
  user_id: string | null;
  actor_name: string | null;
  actor_email: string | null;
  actor_role: string | null;
  actor_type: string;
  action: string;
  module: string;
  resource_type: string | null;
  resource_id: string | null;
  description: string | null;
  status: "success" | "failed";
  event_at: string;
  correlation_id: string | null;
  item_count: number | null;
  resource_ids: string[];
  old_values: Record<string, unknown> | null;
  new_values: Record<string, unknown> | null;
  metadata: Record<string, unknown> | null;
  links: AuditResourceLink[];
}

export interface AuditPage {
  items: AuditEvent[];
  page: number;
  page_size: number;
  total: number;
  pages: number;
}

export interface AuditOptions {
  actions: string[];
  modules: string[];
  roles: string[];
  statuses: string[];
}

export interface AuditFilters {
  page?: number;
  page_size?: number;
  search?: string;
  date_from?: string;
  date_to?: string;
  user_id?: string;
  role?: string;
  entreprise_id?: string;
  module?: string;
  action?: string;
  status?: string;
  sort?: "asc" | "desc";
}
